cmd/misc/commands/validator_stats_partition.go

```diff
@@ -0,0 +1,1153 @@
+}
+
+func (s *statsMigratorConfig) partitionStatsTable(currentTableName, destinationTableName string, numberOfPartitions int) error {
+	tableDefFuncExists, err := existsPGGetTableDef(db.WriterDb)
+	if err != nil {
+		return errors.Wrap(err, "error checking if pg_get_tabledef function exists")
+	}
+
+	if !tableDefFuncExists {
+		logrus.Infof("pg_get_tabledef function does not exist, installing it now")
+		err = installPGGetTableDef(db.WriterDb)
+		if err != nil {
+			return errors.Wrap(err, "error installing pg_get_tabledef function")
+		}
+	}
+
+	if !s.DryRun {
+		if s.DropExisting {
+			_, err := db.WriterDb.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s;", destinationTableName))
+			if err != nil {
+				return errors.Wrap(err, "error dropping existing partitioned table")
+			}
+		} else {
+			var exists bool
+			err := db.WriterDb.Get(&exists, fmt.Sprintf("SELECT EXISTS (SELECT 1 FROM pg_tables WHERE schemaname = 'public' AND tablename = '%s');", destinationTableName))
+			if err != nil {
+				return errors.Wrap(err, "error checking if destination table exists")
+			}
+			if exists {
+				return errors.New("destination table already exists, use --drop-existing if you want to override it")
+			}
+		}
+
+		if s.RenameDestinationOnComplete {
+			var exists bool
+			err := db.WriterDb.Get(&exists, fmt.Sprintf("SELECT EXISTS (SELECT 1 FROM pg_tables WHERE schemaname = 'public' AND tablename = '%s_backup');", currentTableName))
+			if err != nil {
+				return errors.Wrap(err, "error checking if destination table exists")
+			}
+			if exists {
+				return errors.New("backup table of current table already exists, please rename or drop it before running this command")
+			}
+		}
+	}
+
+	logrus.Infof("Part 1: Creating schemas")
+	err = s.createValidatorStatsPartionedTableSchemav1(currentTableName, destinationTableName, numberOfPartitions)
+	if err != nil {
+		return errors.Wrap(err, "error while creating a partitioned table")
+	}
+
+	logrus.Infof("Part 2: Schema creation completed, moving data now")
+
+	// Data moving
+	err = s.copyValidatorStats(destinationTableName, currentTableName, 0)
+	if err != nil {
+		return errors.Wrap(err, "error copying data")
+	}
+
+	// Renaming
+	if s.RenameDestinationOnComplete && !s.DryRun {
+		logrus.Infof("Part 3: Renaming destination table to current table name")
+		err = tableRenaming(currentTableName, destinationTableName, numberOfPartitions)
+		if err != nil {
+			if err == rowMissmatchErr {
+				// This should handle the case when we switch days after data has been copied and before renaming. Remember that only completed days are exported by the statistics exporter.
+				logrus.Infof("Missmatch between current table and destination table row amount for the last exported day. Retrying to export last day.")
+				maxDay := int64(0)
+				err = db.WriterDb.Get(&maxDay, fmt.Sprintf("SELECT max(day) FROM %s", currentTableName))
```

Comment:
The intent here is to re-export the last day only since this handles the edge case of a day switch happening in between. But you are right destinationTableName is probably the better choice just so that it works with multiple days as well if we need to at some point

Reply:
Please also update the info log to say that we retry the export of all missing days.